ROADMAP.md

```diff
@@ -1,62 +1,191 @@
+
+Test suites are run in libp2p/test-plans CI.
+
+**Note:** node.js-libp2p is not listed here since it does not have support in Testground yet.
+
+#### 3. Rock solid libp2p releases
+<!-- TODO: Create issue -->
+The full `libp2p/test-plans` test suite is run before releasing a version of go-libp2p,  rust-libp2p, and js-libp2p.
+A GitHub workflow is added so that these suites run for any PR against master and run against the `master` branch on a nightly basis (updating the status check.)
+
+#### 4. Canonical interop tests & dashboard
+<!-- TODO: Create issue -->
+A comprehensive and canonical dashboard is generated and hosted in a publicly discoverable site that displays latest test suite results (Pass/Fail/Implemented/Not Implemented/Unimplementable) from a nightly CI run.
+All permutations of libp2p implementations, protocols, and versions should be visible.
+
+An enhancement of A.1 to make it easier for users and implementers to see the full state of libp2p interoperability.
+
+### B. Hardening test infrastructure
+
+#### 1. Track test suite stability
+<!-- TODO: Assign a quarter -->
+<!-- TODO: Create issue -->
+`libp2p/test-plans` maintainers have a straightforward way to track the test suite stability and performances
+- We can track the status of every test combination stability from the interop project itself
+- We can easily measure the consequence (improvements) of a pull request to the libp2p/interop repository
+- We are alerted when an interop test starts failing on one of our client repositories, and we can dispatch the alert to repo maintainers.
+
+#### 2. Design process for adding new tests
+<!-- TODO: Assign a quarter -->
+<!-- TODO: Create issue -->
+We have an explicit, working, Design Process for adding new tests
+- The design is documented in `./DESIGN.md`.
+- The design is followed by the team when we add new features.
+- There is a clear path when it comes to testing new features. This might mean testing multiple `masters` against each other.
+
+#### 3. Be the home for all interop tests
+<!-- TODO: Assign a quarter -->
+<!-- TODO: Create issue -->
+We have ported the tests from `libp2p/interop`
+- This repository implement tests `connect`, `dht`, `pubsub`, `steam` ([ref](https://github.com/libp2p/interop/blob/ce0aa3749c9c53cf5ad53009b273847b94106d40/src/index.ts#L32-L35))
+- At of writing (2022-09-27), it is disabled in `go-libp2p` ([ref](https://github.com/libp2p/go-libp2p/actions/workflows/interop.yml)), and it is used in `js-libp2p` ([ref](https://github.com/libp2p/js-libp2p/actions/runs/3111413168/jobs/5050929689)).
+
+#### 4. Stable build process
+<!-- TODO: Assign a quarter -->
+<!-- TODO: Create issue -->
+We have a more stable build process that doesn't risk breaking
+- We generate artifacts for old versions during merges to the libp2p repositories https://github.com/libp2p/test-plans/issues/35#issuecomment-1254991985
```

Comment:
I'm not sure if a stable build process deserves its own point. This seems like the bare minimum that's needed for testing anyway.

Generating artifacts, if I understand correctly, would just be an optimization to speed up build times.

Reply:
I'm not 100% sure whether to keep or remove this.
@laurentsenta do you have any suggestions?

Reply:
@marten-seemann has a point, no strong feelings against removing these